diff --git a/tmrg_files/croc_domainTMR.sv b/tmrg_files/croc_domainTMR.sv
index ec159bc..c3fffbd 100644
--- a/tmrg_files/croc_domainTMR.sv
+++ b/tmrg_files/croc_domainTMR.sv
@@ -38,7 +38,7 @@
  *           MD5 hash          : 3f3a8b3aea489e1bf3c5f7dd5a8b92bc                                   *
  *                                                                                                  *
  ****************************************************************************************************/
-
+`include "common_cells/registers.svh"
 module croc_domainTMR import croc_pkg::*; #(
   parameter int unsigned GpioCount = 16
 ) (
@@ -108,83 +108,76 @@ module croc_domainTMR import croc_pkg::*; #(
   output logic core_busy_oA,
   output logic core_busy_oB,
   output logic core_busy_oC,
-  output tmrError,
   output tmrErrorA,
   output tmrErrorB,
-  output tmrErrorC
+  output tmrErrorC,
+
+  input logic faults_i
 );
+// -----------------
+// Fault signals
+// -----------------
+fault_monitor_reg_pkg::fault_monitor__in_t fm_hwif_in, fm_hwif_in_reg;
+logic [2:0] gpio_faults;
+logic [2:0] uart_faults;
+logic [2:0] timer_faults;
+logic [1:0][2:0] core_faults;
+logic [4:0][2:0] obi_faults;
+sbr_obi_req_t fm_obi_req;
+sbr_obi_rsp_t fm_obi_rsp;
+
 localparam dm::hartinfo_t HARTINFO = '{zero1: '0, nscratch: 2, zero0: '0, dataaccess: 1'b1, datasize: dm::DataCount, dataaddr: dm::DataAddr};
 wire jtag_tdo_o;
-wire dmi_rst_n;
-wire dmi_resp_valid;
-wire dmi_resp_ready;
-wire dm::dmi_resp_t dmi_resp;
-wire dmi_req_valid;
-wire dmi_req_ready;
-wire dm::dmi_req_t dmi_req;
 wire debug_req;
 wire mgr_obi_req_t dbg_req_obi_req;
 wire sbr_obi_rsp_t dbg_mem_obi_rsp;
-wor i_xbar_errtmrErrorC;
-wor i_uarttmrErrorC;
-wor i_timertmrErrorC;
-wor i_sram_shimtmrErrorC;
-wor i_soc_ctrltmrErrorC;
-wor i_soc_ctrl_translatetmrErrorC;
-wor i_periph_errtmrErrorC;
-wor i_periph_cuttmrErrorC;
-wor i_obi_demuxtmrErrorC;
-wor i_main_xbartmrErrorC;
-wor i_gpiotmrErrorC;
-wor i_core_wraptmrErrorC;
-wor i_xbar_errtmrErrorB;
-wor i_uarttmrErrorB;
-wor i_timertmrErrorB;
-wor i_sram_shimtmrErrorB;
-wor i_soc_ctrltmrErrorB;
-wor i_soc_ctrl_translatetmrErrorB;
-wor i_periph_errtmrErrorB;
-wor i_periph_cuttmrErrorB;
-wor i_obi_demuxtmrErrorB;
-wor i_main_xbartmrErrorB;
-wor i_gpiotmrErrorB;
-wor i_core_wraptmrErrorB;
-wor i_xbar_errtmrErrorA;
-wor i_uarttmrErrorA;
-wor i_timertmrErrorA;
-wor i_sram_shimtmrErrorA;
-wor i_soc_ctrltmrErrorA;
-wor i_soc_ctrl_translatetmrErrorA;
-wor i_periph_errtmrErrorA;
-wor i_periph_cuttmrErrorA;
-wor i_obi_demuxtmrErrorA;
-wor i_main_xbartmrErrorA;
-wor i_gpiotmrErrorA;
-wor i_core_wraptmrErrorA;
-wor testmode_iTmrError;
-wor rst_niTmrError;
-wor jtag_trst_niTmrError;
-wor jtag_tms_iTmrError;
-wor jtag_tdi_iTmrError;
-wor jtag_tck_iTmrError;
-wor hartinfoTmrError;
-wor dmi_rst_nTmrError;
-wor dmi_resp_validTmrError;
-wor dmi_resp_readyTmrError;
-wor dmi_respTmrError;
-wor dmi_req_validTmrError;
-wor dmi_req_readyTmrError;
-wor dmi_reqTmrError;
-wor dbg_req_obi_rspTmrError;
-wor dbg_mem_obi_reqTmrError;
-wor clk_iTmrError;
-wire testmode_i;
-wire rst_ni;
+wire i_xbar_errtmrErrorC;
+wire i_uarttmrErrorC;
+wire i_timertmrErrorC;
+wire [NumSramBanks-1:0] i_sram_shimtmrErrorC;
+wire i_soc_ctrltmrErrorC;
+wire i_soc_ctrl_translatetmrErrorC;
+wire i_periph_errtmrErrorC;
+wire  i_periph_cuttmrErrorC;
+wire i_obi_demuxtmrErrorC;
+wire i_main_xbartmrErrorC;
+wire i_gpiotmrErrorC;
+wire i_core_wraptmrErrorC;
+wire i_xbar_errtmrErrorB;
+wire i_uarttmrErrorB;
+wire i_timertmrErrorB;
+wire [NumSramBanks-1:0] i_sram_shimtmrErrorB;
+wire i_soc_ctrltmrErrorB;
+wire i_soc_ctrl_translatetmrErrorB;
+wire i_periph_errtmrErrorB;
+wire i_periph_cuttmrErrorB;
+wire i_obi_demuxtmrErrorB;
+wire i_main_xbartmrErrorB;
+wire i_gpiotmrErrorB;
+wire i_core_wraptmrErrorB;
+wire i_xbar_errtmrErrorA;
+wire i_uarttmrErrorA;
+wire i_timertmrErrorA;
+wire [NumSramBanks-1:0] i_sram_shimtmrErrorA;
+wire i_soc_ctrltmrErrorA;
+wire i_soc_ctrl_translatetmrErrorA;
+wire i_periph_errtmrErrorA;
+wire i_periph_cuttmrErrorA;
+wire i_obi_demuxtmrErrorA;
+wire i_main_xbartmrErrorA;
+wire i_gpiotmrErrorA;
+wire i_core_wraptmrErrorA;
+wire jtag_trst_niTmrError;
+wire jtag_tms_iTmrError;
+wire jtag_tdi_iTmrError;
+wire jtag_tck_iTmrError;
+wire dbg_req_obi_rspTmrError;
+wire dbg_mem_obi_reqTmrError;
 wire jtag_trst_ni;
 wire jtag_tms_i;
 wire jtag_tdi_i;
 wire jtag_tck_i;
-wire dm::hartinfo_t hartinfo;
+wire dm::hartinfo_t hartinfo = HARTINFO;
 wire dmi_rst_n;
 wire dmi_resp_valid;
 wire dmi_resp_ready;
@@ -194,7 +187,6 @@ wire dmi_req_ready;
 wire dm::dmi_req_t dmi_req;
 wire mgr_obi_rsp_t dbg_req_obi_rsp;
 wire sbr_obi_req_t dbg_mem_obi_req;
-wire clk_i;
 logic sram_implA;
 logic sram_implB;
 logic sram_implC;
@@ -288,12 +280,8 @@ mgr_obi_req_t dbg_req_obi_reqC;
 mgr_obi_rsp_t dbg_req_obi_rspA;
 mgr_obi_rsp_t dbg_req_obi_rspB;
 mgr_obi_rsp_t dbg_req_obi_rspC;
-assign dbg_req_obi_reqA.a.aid = '0;
-assign dbg_req_obi_reqB.a.aid = '0;
-assign dbg_req_obi_reqC.a.aid = '0;
-assign dbg_req_obi_reqA.a.a_optional = '0;
-assign dbg_req_obi_reqB.a.a_optional = '0;
-assign dbg_req_obi_reqC.a.a_optional = '0;
+assign dbg_req_obi_req.a.aid = '0;
+assign dbg_req_obi_req.a.a_optional = '0;
 sbr_obi_req_t [NumXbarSbr - 1:0] all_sbr_obi_reqA;
 sbr_obi_req_t [NumXbarSbr - 1:0] all_sbr_obi_reqB;
 sbr_obi_req_t [NumXbarSbr - 1:0] all_sbr_obi_reqC;
@@ -510,35 +498,14 @@ core_wrapTMR i_core_wrap (
     .tmrErrorB(i_core_wraptmrErrorB),
     .tmrErrorC(i_core_wraptmrErrorC)
   );
-dm::hartinfo_t hartinfoA = HARTINFO;
-dm::hartinfo_t hartinfoB = HARTINFO;
-dm::hartinfo_t hartinfoC = HARTINFO;
-logic dmi_rst_nA;
-logic dmi_rst_nB;
-logic dmi_rst_nC;
-logic dmi_req_validA;
-logic dmi_req_validB;
-logic dmi_req_validC;
-logic dmi_req_readyA;
-logic dmi_req_readyB;
-logic dmi_req_readyC;
-logic dmi_resp_validA;
-logic dmi_resp_validB;
-logic dmi_resp_validC;
-logic dmi_resp_readyA;
-logic dmi_resp_readyB;
-logic dmi_resp_readyC;
-dm::dmi_req_t dmi_reqA;
-dm::dmi_req_t dmi_reqB;
-dm::dmi_req_t dmi_reqC;
-dm::dmi_resp_t dmi_respA;
-dm::dmi_resp_t dmi_respB;
-dm::dmi_resp_t dmi_respC;
+  assign core_faults[1][0] = i_core_wraptmrErrorA;
+  assign core_faults[1][1] = i_core_wraptmrErrorB;
+  assign core_faults[1][2] = i_core_wraptmrErrorC;
 
 dmi_jtag #(.IdcodeValue(PulpJtagIdCode)) i_dmi_jtag (
-    .clk_i(clk_i),
-    .rst_ni(rst_ni),
-    .testmode_i(testmode_i),
+    .clk_i(clk_iA),
+    .rst_ni(rst_niA),
+    .testmode_i(testmode_iA),
     .dmi_rst_no(dmi_rst_n),
     .dmi_req_o(dmi_req),
     .dmi_req_valid_o(dmi_req_valid),
@@ -555,9 +522,9 @@ dmi_jtag #(.IdcodeValue(PulpJtagIdCode)) i_dmi_jtag (
   );
 
 dm_obi_top #(.BusWidth(SbrObiCfg.DataWidth), .IdWidth(SbrObiCfg.IdWidth)) i_dm_top (
-    .clk_i(clk_i),
-    .rst_ni(rst_ni),
-    .testmode_i(testmode_i),
+    .clk_i(clk_iA),
+    .rst_ni(rst_niA),
+    .testmode_i(testmode_iA),
     .ndmreset_o(),
     .dmactive_o(),
     .debug_req_o(debug_req),
@@ -591,12 +558,8 @@ dm_obi_top #(.BusWidth(SbrObiCfg.DataWidth), .IdWidth(SbrObiCfg.IdWidth)) i_dm_t
     .dmi_resp_ready_i(dmi_resp_ready),
     .dmi_resp_o(dmi_resp)
   );
-assign dbg_mem_obi_rspA.r.r_optional = 1'b0;
-assign dbg_mem_obi_rspB.r.r_optional = 1'b0;
-assign dbg_mem_obi_rspC.r.r_optional = 1'b0;
-assign dbg_mem_obi_rspA.r.err = 1'b0;
-assign dbg_mem_obi_rspB.r.err = 1'b0;
-assign dbg_mem_obi_rspC.r.err = 1'b0;
+assign dbg_mem_obi_rsp.r.r_optional = 1'b0;
+assign dbg_mem_obi_rsp.r.err = 1'b0;
 
 obi_xbarTMR #(.SbrPortObiCfg(MgrObiCfg), .MgrPortObiCfg(SbrObiCfg), .sbr_port_obi_req_t(mgr_obi_req_t), .sbr_port_a_chan_t(mgr_obi_a_chan_t), .sbr_port_obi_rsp_t(mgr_obi_rsp_t), .sbr_port_r_chan_t(mgr_obi_r_chan_t), .mgr_port_obi_req_t(sbr_obi_req_t), .mgr_port_obi_rsp_t(sbr_obi_rsp_t), .NumSbrPorts(NumXbarManagers), .NumMgrPorts(NumXbarSbr), .NumMaxTrans(2), .NumAddrRules(NumXbarSbrRules), .addr_map_rule_t(addr_map_rule_t), .UseIdForRouting(1'b0), .Connectivity('1)) i_main_xbar (
     .clk_iA(clk_iA),
@@ -633,6 +596,11 @@ obi_xbarTMR #(.SbrPortObiCfg(MgrObiCfg), .MgrPortObiCfg(SbrObiCfg), .sbr_port_ob
     .tmrErrorB(i_main_xbartmrErrorB),
     .tmrErrorC(i_main_xbartmrErrorC)
   );
+
+  assign obi_faults[0][0] = i_main_xbartmrErrorA;
+  assign obi_faults[0][1] = i_main_xbartmrErrorB;
+  assign obi_faults[0][2] = i_main_xbartmrErrorC;
+
 for(genvar i = 0; i<NumSramBanks; i++)
   begin : gen_sram_bank
 logic bank_reqA;
@@ -697,14 +665,16 @@ logic [SbrObiCfg.DataWidth / 8 - 1:0] bank_beC;
         .rdata_iA(bank_rdataA),
         .rdata_iB(bank_rdataB),
         .rdata_iC(bank_rdataC),
-        .tmrErrorA(i_sram_shimtmrErrorA),
-        .tmrErrorB(i_sram_shimtmrErrorB),
-        .tmrErrorC(i_sram_shimtmrErrorC)
+        .tmrErrorA(i_sram_shimtmrErrorA[i]),
+        .tmrErrorB(i_sram_shimtmrErrorB[i]),
+        .tmrErrorC(i_sram_shimtmrErrorC[i])
       );
     assign bank_word_addrA = bank_byte_addrA[SbrObiCfg.AddrWidth - 1:2];
     assign bank_word_addrB = bank_byte_addrB[SbrObiCfg.AddrWidth - 1:2];
     assign bank_word_addrC = bank_byte_addrC[SbrObiCfg.AddrWidth - 1:2];
 
+    assign fm_hwif_in.sram_fault[i].fault_count.incr = i_sram_shimtmrErrorA[i] | i_sram_shimtmrErrorB[i] | i_sram_shimtmrErrorC[i];
+
     tc_sram_impl #(.NumWords(SramBankNumWords), .DataWidth(32), .NumPorts(1), .Latency(1)) i_sramA (
         .clk_i(clk_iA),
         .rst_ni(rst_niA),
@@ -768,6 +738,10 @@ obi_err_sbrTMR #(.ObiCfg(SbrObiCfg), .obi_req_t(sbr_obi_req_t), .obi_rsp_t(sbr_o
     .tmrErrorB(i_xbar_errtmrErrorB),
     .tmrErrorC(i_xbar_errtmrErrorC)
   );
+
+  assign obi_faults[1][0] = i_xbar_errtmrErrorA;
+  assign obi_faults[1][1] = i_xbar_errtmrErrorB;
+  assign obi_faults[1][2] = i_xbar_errtmrErrorC;
 logic [cf_math_pkg::idx_width(NumPeriphs) - 1:0] periph_idxA;
 logic [cf_math_pkg::idx_width(NumPeriphs) - 1:0] periph_idxB;
 logic [cf_math_pkg::idx_width(NumPeriphs) - 1:0] periph_idxC;
@@ -795,8 +769,11 @@ obi_cutTMR #(.ObiCfg(SbrObiCfg), .obi_a_chan_t(sbr_obi_a_chan_t), .obi_r_chan_t(
     .tmrErrorB(i_periph_cuttmrErrorB),
     .tmrErrorC(i_periph_cuttmrErrorC)
   );
+assign obi_faults[4][0] = i_periph_cuttmrErrorA;
+assign obi_faults[4][1] = i_periph_cuttmrErrorB;
+assign obi_faults[4][2] = i_periph_cuttmrErrorC;
 
-addr_decodeTMR #(.NoIndices(NumPeriphs), .NoRules(NumPeriphRules), .rule_t(addr_map_rule_t), .Napot(1'b0)) i_addr_decode_periphs (
+addr_decodeTMR #(.NoIndices(NumPeriphs), .NoRules(NumPeriphRules), .addr_t ( logic[SbrObiCfg.DataWidth-1:0] ), .rule_t(addr_map_rule_t), .Napot(1'b0)) i_addr_decode_periphs (
     .addr_iA(periph_cut_obi_reqA.a.addr),
     .addr_iB(periph_cut_obi_reqB.a.addr),
     .addr_iC(periph_cut_obi_reqC.a.addr),
@@ -847,6 +824,10 @@ obi_demuxTMR #(.ObiCfg(SbrObiCfg), .obi_req_t(sbr_obi_req_t), .obi_rsp_t(sbr_obi
     .tmrErrorC(i_obi_demuxtmrErrorC)
   );
 
+  assign obi_faults[2][0] = i_obi_demuxtmrErrorA;
+  assign obi_faults[2][1] = i_obi_demuxtmrErrorB;
+  assign obi_faults[2][2] = i_obi_demuxtmrErrorC;
+
 obi_err_sbrTMR #(.ObiCfg(SbrObiCfg), .obi_req_t(sbr_obi_req_t), .obi_rsp_t(sbr_obi_rsp_t), .NumMaxTrans(1), .RspData(32'hBADCAB1E)) i_periph_err (
     .clk_iA(clk_iA),
     .clk_iB(clk_iB),
@@ -867,6 +848,10 @@ obi_err_sbrTMR #(.ObiCfg(SbrObiCfg), .obi_req_t(sbr_obi_req_t), .obi_rsp_t(sbr_o
     .tmrErrorB(i_periph_errtmrErrorB),
     .tmrErrorC(i_periph_errtmrErrorC)
   );
+
+  assign obi_faults[3][0] = i_periph_errtmrErrorA;
+  assign obi_faults[3][1] = i_periph_errtmrErrorB;
+  assign obi_faults[3][2] = i_periph_errtmrErrorC;
 reg_req_t soc_ctrl_reg_reqA;
 reg_req_t soc_ctrl_reg_reqB;
 reg_req_t soc_ctrl_reg_reqC;
@@ -972,6 +957,9 @@ soc_ctrl_reg_topTMR #(.reg_req_t(reg_req_t), .reg_rsp_t(reg_rsp_t), .BootAddrDef
     .tmrErrorB(i_soc_ctrltmrErrorB),
     .tmrErrorC(i_soc_ctrltmrErrorC)
   );
+assign core_faults[0][0] = i_soc_ctrl_translatetmrErrorA | i_soc_ctrltmrErrorA;
+assign core_faults[0][1] = i_soc_ctrl_translatetmrErrorB | i_soc_ctrltmrErrorB;
+assign core_faults[0][2] = i_soc_ctrl_translatetmrErrorC | i_soc_ctrltmrErrorC;
 
 obi_uartTMR #(.ObiCfg(SbrObiCfg), .obi_req_t(sbr_obi_req_t), .obi_rsp_t(sbr_obi_rsp_t)) i_uart (
     .clk_iA(clk_iA),
@@ -1027,6 +1015,10 @@ obi_uartTMR #(.ObiCfg(SbrObiCfg), .obi_req_t(sbr_obi_req_t), .obi_rsp_t(sbr_obi_
     .tmrErrorC(i_uarttmrErrorC)
   );
 
+  assign uart_faults[0] = i_uarttmrErrorA;
+  assign uart_faults[1] = i_uarttmrErrorB;
+  assign uart_faults[2] = i_uarttmrErrorC;
+
 gpioTMR #(.ObiCfg(SbrObiCfg), .obi_req_t(sbr_obi_req_t), .obi_rsp_t(sbr_obi_rsp_t), .GpioCount(GpioCount)) i_gpio (
     .clk_iA(clk_iA),
     .clk_iB(clk_iB),
@@ -1060,6 +1052,10 @@ gpioTMR #(.ObiCfg(SbrObiCfg), .obi_req_t(sbr_obi_req_t), .obi_rsp_t(sbr_obi_rsp_
     .tmrErrorC(i_gpiotmrErrorC)
   );
 
+  assign gpio_faults[0] = i_gpiotmrErrorA;
+  assign gpio_faults[1] = i_gpiotmrErrorB;
+  assign gpio_faults[2] = i_gpiotmrErrorC;
+
 timer_unitTMR #(.ID_WIDTH(SbrObiCfg.IdWidth)) i_timer (
     .clk_iA(clk_iA),
     .clk_iB(clk_iB),
@@ -1122,6 +1118,55 @@ timer_unitTMR #(.ID_WIDTH(SbrObiCfg.IdWidth)) i_timer (
     .tmrErrorB(i_timertmrErrorB),
     .tmrErrorC(i_timertmrErrorC)
   );
+
+  assign timer_faults[0] = i_timertmrErrorA;
+  assign timer_faults[1] = i_timertmrErrorB;
+  assign timer_faults[2] = i_timertmrErrorC;
+
+assign fm_hwif_in.obi_fault.fault_count.incr = (|obi_faults) | jtag_trst_niTmrError | jtag_tms_iTmrError | jtag_tdi_iTmrError | jtag_tck_iTmrError | dbg_req_obi_rspTmrError | dbg_mem_obi_reqTmrError;
+assign fm_hwif_in.gpio_fault.fault_count.incr = |gpio_faults;
+assign fm_hwif_in.uart_fault.fault_count.incr = |uart_faults;
+assign fm_hwif_in.core_fault.fault_count.incr = |core_faults;
+assign fm_hwif_in.soc_fault.fault_count.incr = faults_i;
+assign fm_hwif_in.timer_fault.fault_count.incr = |timer_faults;
+
+periph_to_fm i_periph_to_fm (
+  .periph_req_iA(all_periph_obi_reqA[PeriphFaultMonitor]),
+  .periph_req_iB(all_periph_obi_reqB[PeriphFaultMonitor]),
+  .periph_req_iC(all_periph_obi_reqC[PeriphFaultMonitor]),
+  .fm_req_o(fm_obi_req),
+
+  .fm_rsp_i(fm_obi_rsp),
+  .periph_rsp_oA(all_periph_obi_rspA[PeriphFaultMonitor]),
+  .periph_rsp_oB(all_periph_obi_rspB[PeriphFaultMonitor]),
+  .periph_rsp_oC(all_periph_obi_rspC[PeriphFaultMonitor])
+);
+
+`FF(fm_hwif_in_reg, fm_hwif_in, '{default: '0}, clk_iA, rst_niA);
+
+  fault_monitor_reg_top #(
+    .ID_WIDTH ( SbrObiCfg.IdWidth )
+  ) i_fault_monitor (
+    .clk ( clk_iA ),
+    .arst_n ( rst_niA ),
+
+    .obi_req    ( fm_obi_req.req ),
+    .obi_gnt    ( fm_obi_rsp.gnt ),
+    .obi_addr   ( fm_obi_req.a.addr[4:0] ),
+    .obi_we     ( fm_obi_req.a.we ),
+    .obi_be     ( fm_obi_req.a.be ),
+    .obi_wdata  ( fm_obi_req.a.wdata ),
+    .obi_aid    ( fm_obi_req.a.aid ),
+    .obi_rvalid ( fm_obi_rsp.rvalid ),
+    .obi_rready ( '1 ),
+    .obi_rdata  ( fm_obi_rsp.r.rdata ),
+    .obi_err    ( fm_obi_rsp.r.err ),
+    .obi_rid    ( fm_obi_rsp.r.rid ),
+
+    .hwif_in ( fm_hwif_in_reg )
+  );
+  assign fm_obi_rsp.r.r_optional = '0;
+
 assign timer_obi_rspA.r.err = 1'b0;
 assign timer_obi_rspB.r.err = 1'b0;
 assign timer_obi_rspC.r.err = 1'b0;
@@ -1129,14 +1174,6 @@ assign timer_obi_rspA.r.r_optional = 1'b0;
 assign timer_obi_rspB.r.r_optional = 1'b0;
 assign timer_obi_rspC.r.r_optional = 1'b0;
 
-majorityVoter clk_iVoter (
-    .inA(clk_iA),
-    .inB(clk_iB),
-    .inC(clk_iC),
-    .out(clk_i),
-    .tmrErr(clk_iTmrError)
-  );
-
 majorityVoter #(.WIDTH( ($size(dbg_mem_obi_req)) )) dbg_mem_obi_reqVoter (
     .inA(dbg_mem_obi_reqA),
     .inB(dbg_mem_obi_reqB),
@@ -1153,70 +1190,6 @@ majorityVoter #(.WIDTH( ($size(dbg_req_obi_rsp)) )) dbg_req_obi_rspVoter (
     .tmrErr(dbg_req_obi_rspTmrError)
   );
 
-majorityVoter #(.WIDTH( ($size(dmi_req)) )) dmi_reqVoter (
-    .inA(dmi_reqA),
-    .inB(dmi_reqB),
-    .inC(dmi_reqC),
-    .out(dmi_req),
-    .tmrErr(dmi_reqTmrError)
-  );
-
-majorityVoter dmi_req_readyVoter (
-    .inA(dmi_req_readyA),
-    .inB(dmi_req_readyB),
-    .inC(dmi_req_readyC),
-    .out(dmi_req_ready),
-    .tmrErr(dmi_req_readyTmrError)
-  );
-
-majorityVoter dmi_req_validVoter (
-    .inA(dmi_req_validA),
-    .inB(dmi_req_validB),
-    .inC(dmi_req_validC),
-    .out(dmi_req_valid),
-    .tmrErr(dmi_req_validTmrError)
-  );
-
-majorityVoter #(.WIDTH( ($size(dmi_resp)) )) dmi_respVoter (
-    .inA(dmi_respA),
-    .inB(dmi_respB),
-    .inC(dmi_respC),
-    .out(dmi_resp),
-    .tmrErr(dmi_respTmrError)
-  );
-
-majorityVoter dmi_resp_readyVoter (
-    .inA(dmi_resp_readyA),
-    .inB(dmi_resp_readyB),
-    .inC(dmi_resp_readyC),
-    .out(dmi_resp_ready),
-    .tmrErr(dmi_resp_readyTmrError)
-  );
-
-majorityVoter dmi_resp_validVoter (
-    .inA(dmi_resp_validA),
-    .inB(dmi_resp_validB),
-    .inC(dmi_resp_validC),
-    .out(dmi_resp_valid),
-    .tmrErr(dmi_resp_validTmrError)
-  );
-
-majorityVoter dmi_rst_nVoter (
-    .inA(dmi_rst_nA),
-    .inB(dmi_rst_nB),
-    .inC(dmi_rst_nC),
-    .out(dmi_rst_n),
-    .tmrErr(dmi_rst_nTmrError)
-  );
-
-majorityVoter #(.WIDTH( ($size(hartinfo)) )) hartinfoVoter (
-    .inA(hartinfoA),
-    .inB(hartinfoB),
-    .inC(hartinfoC),
-    .out(hartinfo),
-    .tmrErr(hartinfoTmrError)
-  );
-
 majorityVoter jtag_tck_iVoter (
     .inA(jtag_tck_iA),
     .inB(jtag_tck_iB),
@@ -1249,25 +1222,9 @@ majorityVoter jtag_trst_niVoter (
     .tmrErr(jtag_trst_niTmrError)
   );
 
-majorityVoter rst_niVoter (
-    .inA(rst_niA),
-    .inB(rst_niB),
-    .inC(rst_niC),
-    .out(rst_ni),
-    .tmrErr(rst_niTmrError)
-  );
-
-majorityVoter testmode_iVoter (
-    .inA(testmode_iA),
-    .inB(testmode_iB),
-    .inC(testmode_iC),
-    .out(testmode_i),
-    .tmrErr(testmode_iTmrError)
-  );
-assign tmrError = clk_iTmrError|dbg_mem_obi_reqTmrError|dbg_req_obi_rspTmrError|dmi_reqTmrError|dmi_req_readyTmrError|dmi_req_validTmrError|dmi_respTmrError|dmi_resp_readyTmrError|dmi_resp_validTmrError|dmi_rst_nTmrError|hartinfoTmrError|jtag_tck_iTmrError|jtag_tdi_iTmrError|jtag_tms_iTmrError|jtag_trst_niTmrError|rst_niTmrError|testmode_iTmrError;
-assign tmrErrorA = i_core_wraptmrErrorA|i_gpiotmrErrorA|i_main_xbartmrErrorA|i_obi_demuxtmrErrorA|i_periph_cuttmrErrorA|i_periph_errtmrErrorA|i_soc_ctrl_translatetmrErrorA|i_soc_ctrltmrErrorA|i_sram_shimtmrErrorA|i_timertmrErrorA|i_uarttmrErrorA|i_xbar_errtmrErrorA;
-assign tmrErrorB = i_core_wraptmrErrorB|i_gpiotmrErrorB|i_main_xbartmrErrorB|i_obi_demuxtmrErrorB|i_periph_cuttmrErrorB|i_periph_errtmrErrorB|i_soc_ctrl_translatetmrErrorB|i_soc_ctrltmrErrorB|i_sram_shimtmrErrorB|i_timertmrErrorB|i_uarttmrErrorB|i_xbar_errtmrErrorB;
-assign tmrErrorC = i_core_wraptmrErrorC|i_gpiotmrErrorC|i_main_xbartmrErrorC|i_obi_demuxtmrErrorC|i_periph_cuttmrErrorC|i_periph_errtmrErrorC|i_soc_ctrl_translatetmrErrorC|i_soc_ctrltmrErrorC|i_sram_shimtmrErrorC|i_timertmrErrorC|i_uarttmrErrorC|i_xbar_errtmrErrorC;
+assign tmrErrorA = i_core_wraptmrErrorA|i_gpiotmrErrorA|i_main_xbartmrErrorA|i_obi_demuxtmrErrorA|i_periph_cuttmrErrorA|i_periph_errtmrErrorA|i_soc_ctrl_translatetmrErrorA|i_soc_ctrltmrErrorA|(|i_sram_shimtmrErrorA)|i_timertmrErrorA|i_uarttmrErrorA|i_xbar_errtmrErrorA;
+assign tmrErrorB = i_core_wraptmrErrorB|i_gpiotmrErrorB|i_main_xbartmrErrorB|i_obi_demuxtmrErrorB|i_periph_cuttmrErrorB|i_periph_errtmrErrorB|i_soc_ctrl_translatetmrErrorB|i_soc_ctrltmrErrorB|(|i_sram_shimtmrErrorB)|i_timertmrErrorB|i_uarttmrErrorB|i_xbar_errtmrErrorB;
+assign tmrErrorC = i_core_wraptmrErrorC|i_gpiotmrErrorC|i_main_xbartmrErrorC|i_obi_demuxtmrErrorC|i_periph_cuttmrErrorC|i_periph_errtmrErrorC|i_soc_ctrl_translatetmrErrorC|i_soc_ctrltmrErrorC|(|i_sram_shimtmrErrorC)|i_timertmrErrorC|i_uarttmrErrorC|i_xbar_errtmrErrorC;
 
 fanout #(.WIDTH( ($size(dbg_mem_obi_rsp)) )) dbg_mem_obi_rspFanout (
     .in(dbg_mem_obi_rsp),
@@ -1290,55 +1247,6 @@ fanout debug_reqFanout (
     .outC(debug_reqC)
   );
 
-fanout #(.WIDTH( ($size(dmi_req)) )) dmi_reqFanout (
-    .in(dmi_req),
-    .outA(dmi_reqA),
-    .outB(dmi_reqB),
-    .outC(dmi_reqC)
-  );
-
-fanout dmi_req_readyFanout (
-    .in(dmi_req_ready),
-    .outA(dmi_req_readyA),
-    .outB(dmi_req_readyB),
-    .outC(dmi_req_readyC)
-  );
-
-fanout dmi_req_validFanout (
-    .in(dmi_req_valid),
-    .outA(dmi_req_validA),
-    .outB(dmi_req_validB),
-    .outC(dmi_req_validC)
-  );
-
-fanout #(.WIDTH( ($size(dmi_resp)) )) dmi_respFanout (
-    .in(dmi_resp),
-    .outA(dmi_respA),
-    .outB(dmi_respB),
-    .outC(dmi_respC)
-  );
-
-fanout dmi_resp_readyFanout (
-    .in(dmi_resp_ready),
-    .outA(dmi_resp_readyA),
-    .outB(dmi_resp_readyB),
-    .outC(dmi_resp_readyC)
-  );
-
-fanout dmi_resp_validFanout (
-    .in(dmi_resp_valid),
-    .outA(dmi_resp_validA),
-    .outB(dmi_resp_validB),
-    .outC(dmi_resp_validC)
-  );
-
-fanout dmi_rst_nFanout (
-    .in(dmi_rst_n),
-    .outA(dmi_rst_nA),
-    .outB(dmi_rst_nB),
-    .outC(dmi_rst_nC)
-  );
-
 fanout jtag_tdo_oFanout (
     .in(jtag_tdo_o),
     .outA(jtag_tdo_oA),
