diff --git a/tmrg_files/cve2_multdiv_fastTMR.sv b/tmrg_files/cve2_multdiv_fastTMR.sv
index 3c16585..76afddd 100644
--- a/tmrg_files/cve2_multdiv_fastTMR.sv
+++ b/tmrg_files/cve2_multdiv_fastTMR.sv
@@ -109,31 +109,27 @@ module cve2_multdiv_fastTMR #(
   output tmrErrorC
 );
 import cve2_pkg::*;
-parameter MD_IDLE = 3'b000;
-parameter MD_ABS_A = 3'b001;
-parameter MD_ABS_B = 3'b010;
-parameter MD_COMP = 3'b011;
-parameter MD_LAST = 3'b100;
-parameter MD_CHANGE_SIGN = 3'b101;
-parameter MD_FINISH = 3'b110;
-wor op_quotient_qTmrErrorC;
-wor op_numerator_qTmrErrorC;
-wor mult_state_qTmrErrorC;
-wor md_state_qTmrErrorC;
-wor div_counter_qTmrErrorC;
-wor div_by_zero_qTmrErrorC;
-wor op_quotient_qTmrErrorB;
-wor op_numerator_qTmrErrorB;
-wor mult_state_qTmrErrorB;
-wor md_state_qTmrErrorB;
-wor div_counter_qTmrErrorB;
-wor div_by_zero_qTmrErrorB;
-wor op_quotient_qTmrErrorA;
-wor op_numerator_qTmrErrorA;
-wor mult_state_qTmrErrorA;
-wor md_state_qTmrErrorA;
-wor div_counter_qTmrErrorA;
-wor div_by_zero_qTmrErrorA;
+typedef enum logic [2:0] {
+  MD_IDLE, MD_ABS_A, MD_ABS_B, MD_COMP, MD_LAST, MD_CHANGE_SIGN, MD_FINISH
+} md_fsm_e;
+wire op_quotient_qTmrErrorC;
+wire op_numerator_qTmrErrorC;
+wire mult_state_qTmrErrorC;
+wire md_state_qTmrErrorC;
+wire div_counter_qTmrErrorC;
+wire div_by_zero_qTmrErrorC;
+wire op_quotient_qTmrErrorB;
+wire op_numerator_qTmrErrorB;
+wire mult_state_qTmrErrorB;
+wire md_state_qTmrErrorB;
+wire div_counter_qTmrErrorB;
+wire div_by_zero_qTmrErrorB;
+wire op_quotient_qTmrErrorA;
+wire op_numerator_qTmrErrorA;
+wire mult_state_qTmrErrorA;
+wire md_state_qTmrErrorA;
+wire div_counter_qTmrErrorA;
+wire div_by_zero_qTmrErrorA;
 logic signed [34:0] mac_res_signedA;
 logic signed [34:0] mac_res_signedB;
 logic signed [34:0] mac_res_signedC;
@@ -251,15 +247,9 @@ logic mult_en_internalC;
 logic div_en_internalA;
 logic div_en_internalB;
 logic div_en_internalC;
-reg [2:0] md_state_qA;
-reg [2:0] md_state_qB;
-reg [2:0] md_state_qC;
-reg [2:0] md_state_dA;
-reg [2:0] md_state_dB;
-reg [2:0] md_state_dC;
-reg [2:0] md_state_qVotedA;
-reg [2:0] md_state_qVotedB;
-reg [2:0] md_state_qVotedC;
+md_fsm_e md_state_qA, md_state_dA, md_state_qVotedA;
+md_fsm_e md_state_qB, md_state_dB, md_state_qVotedB;
+md_fsm_e md_state_qC, md_state_dC, md_state_qVotedC;
 logic unused_mult_sel_iA;
 logic unused_mult_sel_iB;
 logic unused_mult_sel_iC;
