diff --git a/tmrg_files/cve2_prefetch_bufferTMR.sv b/tmrg_files/cve2_prefetch_bufferTMR.sv
index beb8e25..8405e2a 100644
--- a/tmrg_files/cve2_prefetch_bufferTMR.sv
+++ b/tmrg_files/cve2_prefetch_bufferTMR.sv
@@ -99,27 +99,27 @@ module cve2_prefetch_bufferTMR (
   output tmrErrorC
 );
 localparam int unsigned NUM_REQS = 2;
-wor valid_req_qTmrErrorC;
-wor stored_addr_qTmrErrorC;
-wor rdata_outstanding_qTmrErrorC;
-wor fifo_itmrErrorC;
-wor fetch_addr_qTmrErrorC;
-wor discard_req_qTmrErrorC;
-wor branch_discard_qTmrErrorC;
-wor valid_req_qTmrErrorB;
-wor stored_addr_qTmrErrorB;
-wor rdata_outstanding_qTmrErrorB;
-wor fifo_itmrErrorB;
-wor fetch_addr_qTmrErrorB;
-wor discard_req_qTmrErrorB;
-wor branch_discard_qTmrErrorB;
-wor valid_req_qTmrErrorA;
-wor stored_addr_qTmrErrorA;
-wor rdata_outstanding_qTmrErrorA;
-wor fifo_itmrErrorA;
-wor fetch_addr_qTmrErrorA;
-wor discard_req_qTmrErrorA;
-wor branch_discard_qTmrErrorA;
+wire valid_req_qTmrErrorC;
+wire stored_addr_qTmrErrorC;
+wire rdata_outstanding_qTmrErrorC;
+wire fifo_itmrErrorC;
+wire fetch_addr_qTmrErrorC;
+wire discard_req_qTmrErrorC;
+wire branch_discard_qTmrErrorC;
+wire valid_req_qTmrErrorB;
+wire stored_addr_qTmrErrorB;
+wire rdata_outstanding_qTmrErrorB;
+wire fifo_itmrErrorB;
+wire fetch_addr_qTmrErrorB;
+wire discard_req_qTmrErrorB;
+wire branch_discard_qTmrErrorB;
+wire valid_req_qTmrErrorA;
+wire stored_addr_qTmrErrorA;
+wire rdata_outstanding_qTmrErrorA;
+wire fifo_itmrErrorA;
+wire fetch_addr_qTmrErrorA;
+wire discard_req_qTmrErrorA;
+wire branch_discard_qTmrErrorA;
 logic valid_new_reqA;
 logic valid_new_reqB;
 logic valid_new_reqC;
