diff --git a/tmrg_files/cve2_controllerTMR.sv b/tmrg_files/cve2_controllerTMR.sv
index 3b1e40e..95b777a 100644
--- a/tmrg_files/cve2_controllerTMR.sv
+++ b/tmrg_files/cve2_controllerTMR.sv
@@ -213,52 +213,40 @@ module cve2_controllerTMR (
   output tmrErrorC
 );
 import cve2_pkg::*;
-parameter RESET = 4'h0;
-parameter BOOT_SET = 4'h1;
-parameter WAIT_SLEEP = 4'h2;
-parameter SLEEP = 4'h3;
-parameter FIRST_FETCH = 4'h4;
-parameter DECODE = 4'h5;
-parameter FLUSH = 4'h6;
-parameter IRQ_TAKEN = 4'h7;
-parameter DBG_TAKEN_IF = 4'h8;
-parameter DBG_TAKEN_ID = 4'h9;
-wor store_err_qTmrErrorC;
-wor nmi_mode_qTmrErrorC;
-wor load_err_qTmrErrorC;
-wor illegal_insn_qTmrErrorC;
-wor exc_req_qTmrErrorC;
-wor enter_debug_mode_prio_qTmrErrorC;
-wor do_single_step_qTmrErrorC;
-wor debug_mode_qTmrErrorC;
-wor ctrl_fsm_csTmrErrorC;
-wor store_err_qTmrErrorB;
-wor nmi_mode_qTmrErrorB;
-wor load_err_qTmrErrorB;
-wor illegal_insn_qTmrErrorB;
-wor exc_req_qTmrErrorB;
-wor enter_debug_mode_prio_qTmrErrorB;
-wor do_single_step_qTmrErrorB;
-wor debug_mode_qTmrErrorB;
-wor ctrl_fsm_csTmrErrorB;
-wor store_err_qTmrErrorA;
-wor nmi_mode_qTmrErrorA;
-wor load_err_qTmrErrorA;
-wor illegal_insn_qTmrErrorA;
-wor exc_req_qTmrErrorA;
-wor enter_debug_mode_prio_qTmrErrorA;
-wor do_single_step_qTmrErrorA;
-wor debug_mode_qTmrErrorA;
-wor ctrl_fsm_csTmrErrorA;
-reg [3:0] ctrl_fsm_csA;
-reg [3:0] ctrl_fsm_csB;
-reg [3:0] ctrl_fsm_csC;
-reg [3:0] ctrl_fsm_nsA;
-reg [3:0] ctrl_fsm_nsB;
-reg [3:0] ctrl_fsm_nsC;
-reg [3:0] ctrl_fsm_csVotedA;
-reg [3:0] ctrl_fsm_csVotedB;
-reg [3:0] ctrl_fsm_csVotedC;
+typedef enum logic [3:0] {
+  RESET, BOOT_SET, WAIT_SLEEP, SLEEP, FIRST_FETCH, DECODE, FLUSH,
+  IRQ_TAKEN, DBG_TAKEN_IF, DBG_TAKEN_ID
+} ctrl_fsm_e;
+wire store_err_qTmrErrorC;
+wire nmi_mode_qTmrErrorC;
+wire load_err_qTmrErrorC;
+wire illegal_insn_qTmrErrorC;
+wire exc_req_qTmrErrorC;
+wire enter_debug_mode_prio_qTmrErrorC;
+wire do_single_step_qTmrErrorC;
+wire debug_mode_qTmrErrorC;
+wire ctrl_fsm_csTmrErrorC;
+wire store_err_qTmrErrorB;
+wire nmi_mode_qTmrErrorB;
+wire load_err_qTmrErrorB;
+wire illegal_insn_qTmrErrorB;
+wire exc_req_qTmrErrorB;
+wire enter_debug_mode_prio_qTmrErrorB;
+wire do_single_step_qTmrErrorB;
+wire debug_mode_qTmrErrorB;
+wire ctrl_fsm_csTmrErrorB;
+wire store_err_qTmrErrorA;
+wire nmi_mode_qTmrErrorA;
+wire load_err_qTmrErrorA;
+wire illegal_insn_qTmrErrorA;
+wire exc_req_qTmrErrorA;
+wire enter_debug_mode_prio_qTmrErrorA;
+wire do_single_step_qTmrErrorA;
+wire debug_mode_qTmrErrorA;
+wire ctrl_fsm_csTmrErrorA;
+ctrl_fsm_e ctrl_fsm_csA, ctrl_fsm_nsA, ctrl_fsm_csVotedA;
+ctrl_fsm_e ctrl_fsm_csB, ctrl_fsm_nsB, ctrl_fsm_csVotedB;
+ctrl_fsm_e ctrl_fsm_csC, ctrl_fsm_nsC, ctrl_fsm_csVotedC;
 logic nmi_mode_qA;
 logic nmi_mode_qB;
 logic nmi_mode_qC;
@@ -1659,7 +1647,7 @@ majorityVoter #(.WIDTH(4)) ctrl_fsm_csVoterA (
     .inA(ctrl_fsm_csA),
     .inB(ctrl_fsm_csB),
     .inC(ctrl_fsm_csC),
-    .out(ctrl_fsm_csVotedA),
+    .out({ctrl_fsm_csVotedA}),
     .tmrErr(ctrl_fsm_csTmrErrorA)
   );
 
@@ -1732,7 +1720,7 @@ majorityVoter #(.WIDTH(4)) ctrl_fsm_csVoterB (
     .inA(ctrl_fsm_csA),
     .inB(ctrl_fsm_csB),
     .inC(ctrl_fsm_csC),
-    .out(ctrl_fsm_csVotedB),
+    .out({ctrl_fsm_csVotedB}),
     .tmrErr(ctrl_fsm_csTmrErrorB)
   );
 
@@ -1805,7 +1793,7 @@ majorityVoter #(.WIDTH(4)) ctrl_fsm_csVoterC (
     .inA(ctrl_fsm_csA),
     .inB(ctrl_fsm_csB),
     .inC(ctrl_fsm_csC),
-    .out(ctrl_fsm_csVotedC),
+    .out({ctrl_fsm_csVotedC}),
     .tmrErr(ctrl_fsm_csTmrErrorC)
   );
 
