diff --git a/tmrg_files/gpioTMR.sv b/tmrg_files/gpioTMR.sv
index a5dd11f..88353d1 100644
--- a/tmrg_files/gpioTMR.sv
+++ b/tmrg_files/gpioTMR.sv
@@ -42,6 +42,8 @@
 `include "common_cells/registers.svh"
 module gpioTMR #(
   parameter obi_pkg::obi_cfg_t ObiCfg = obi_pkg::ObiDefaultConfig,
+  parameter type obi_req_t         = logic,
+  parameter type obi_rsp_t         = logic,
   parameter int NrSyncStages = 2,
   parameter int unsigned GpioCount = 16
 ) (
@@ -77,15 +79,15 @@ module gpioTMR #(
   output tmrErrorC
 );
 import gpio_reg_pkg::*;
-wor serial_qTmrErrorC;
-wor i_synctmrErrorC;
-wor i_reg_filetmrErrorC;
-wor serial_qTmrErrorB;
-wor i_synctmrErrorB;
-wor i_reg_filetmrErrorB;
-wor serial_qTmrErrorA;
-wor i_synctmrErrorA;
-wor i_reg_filetmrErrorA;
+wire [GpioCount-1:0] serial_qTmrErrorC;
+wire [GpioCount-1:0] i_synctmrErrorC;
+wire i_reg_filetmrErrorC;
+wire [GpioCount-1:0] serial_qTmrErrorB;
+wire [GpioCount-1:0] i_synctmrErrorB;
+wire i_reg_filetmrErrorB;
+wire [GpioCount-1:0] serial_qTmrErrorA;
+wire [GpioCount-1:0] i_synctmrErrorA;
+wire i_reg_filetmrErrorA;
 gpio_reg2hw_t [GpioCount - 1:0] reg2hwA;
 gpio_reg2hw_t [GpioCount - 1:0] reg2hwB;
 gpio_reg2hw_t [GpioCount - 1:0] reg2hwC;
@@ -162,9 +164,9 @@ logic serial_qVotedC;
         .serial_oA(serial_dA),
         .serial_oB(serial_dB),
         .serial_oC(serial_dC),
-        .tmrErrorA(i_synctmrErrorA),
-        .tmrErrorB(i_synctmrErrorB),
-        .tmrErrorC(i_synctmrErrorC)
+        .tmrErrorA(i_synctmrErrorA[idx]),
+        .tmrErrorB(i_synctmrErrorB[idx]),
+        .tmrErrorC(i_synctmrErrorC[idx])
       );
 
     always_ff @( posedge clk_iA or negedge rst_niA )
@@ -280,6 +282,30 @@ logic is_outputC;
     assign hw2regA[idx].intrpt_valid = gpio_intrptA[idx];
     assign hw2regB[idx].intrpt_valid = gpio_intrptB[idx];
     assign hw2regC[idx].intrpt_valid = gpio_intrptC[idx];
+
+    majorityVoter serial_qVoterA (
+      .inA(serial_qA),
+      .inB(serial_qB),
+      .inC(serial_qC),
+      .out(serial_qVotedA),
+      .tmrErr(serial_qTmrErrorA[idx])
+    );
+
+    majorityVoter serial_qVoterB (
+      .inA(serial_qA),
+      .inB(serial_qB),
+      .inC(serial_qC),
+      .out(serial_qVotedB),
+      .tmrErr(serial_qTmrErrorB[idx])
+    );
+
+    majorityVoter serial_qVoterC (
+        .inA(serial_qA),
+        .inB(serial_qB),
+        .inC(serial_qC),
+        .out(serial_qVotedC),
+        .tmrErr(serial_qTmrErrorC[idx])
+      );
   end
 
 always_comb
@@ -312,31 +338,10 @@ assign interrupt_oA = gpio_intrpt_pendingA;
 assign interrupt_oB = gpio_intrpt_pendingB;
 assign interrupt_oC = gpio_intrpt_pendingC;
 
-majorityVoter serial_qVoterA (
-    .inA(serial_qA),
-    .inB(serial_qB),
-    .inC(serial_qC),
-    .out(serial_qVotedA),
-    .tmrErr(serial_qTmrErrorA)
-  );
-assign tmrErrorA = i_reg_filetmrErrorA|i_synctmrErrorA|serial_qTmrErrorA;
+assign tmrErrorA = i_reg_filetmrErrorA|(|i_synctmrErrorA)|(|serial_qTmrErrorA);
 
-majorityVoter serial_qVoterB (
-    .inA(serial_qA),
-    .inB(serial_qB),
-    .inC(serial_qC),
-    .out(serial_qVotedB),
-    .tmrErr(serial_qTmrErrorB)
-  );
-assign tmrErrorB = i_reg_filetmrErrorB|i_synctmrErrorB|serial_qTmrErrorB;
+assign tmrErrorB = i_reg_filetmrErrorB|(|i_synctmrErrorB)|(|serial_qTmrErrorB);
 
-majorityVoter serial_qVoterC (
-    .inA(serial_qA),
-    .inB(serial_qB),
-    .inC(serial_qC),
-    .out(serial_qVotedC),
-    .tmrErr(serial_qTmrErrorC)
-  );
-assign tmrErrorC = i_reg_filetmrErrorC|i_synctmrErrorC|serial_qTmrErrorC;
+assign tmrErrorC = i_reg_filetmrErrorC|(|i_synctmrErrorC)|(|serial_qTmrErrorC);
 endmodule
 
