diff --git a/tmrg_files/obi_uart_registerTMR.sv b/tmrg_files/obi_uart_registerTMR.sv
index c49064a..8a0dcd2 100644
--- a/tmrg_files/obi_uart_registerTMR.sv
+++ b/tmrg_files/obi_uart_registerTMR.sv
@@ -41,7 +41,9 @@
 
 `include "common_cells/registers.svh"
 module obi_uart_registerTMR import obi_uart_pkg::*; #(
-  parameter obi_pkg::obi_cfg_t ObiCfg = obi_pkg::ObiDefaultConfig
+  parameter obi_pkg::obi_cfg_t ObiCfg = obi_pkg::ObiDefaultConfig,
+  parameter type               obi_req_t = logic,
+  parameter type               obi_rsp_t = logic
 ) (
   input logic clk_iA,
   input logic clk_iB,
@@ -65,27 +67,27 @@ module obi_uart_registerTMR import obi_uart_pkg::*; #(
   output tmrErrorB,
   output tmrErrorC
 );
-wor word_addr_qTmrErrorC;
-wor we_qTmrErrorC;
-wor w_err_qTmrErrorC;
-wor valid_qTmrErrorC;
-wor req_qTmrErrorC;
-wor reg_qTmrErrorC;
-wor id_qTmrErrorC;
-wor word_addr_qTmrErrorB;
-wor we_qTmrErrorB;
-wor w_err_qTmrErrorB;
-wor valid_qTmrErrorB;
-wor req_qTmrErrorB;
-wor reg_qTmrErrorB;
-wor id_qTmrErrorB;
-wor word_addr_qTmrErrorA;
-wor we_qTmrErrorA;
-wor w_err_qTmrErrorA;
-wor valid_qTmrErrorA;
-wor req_qTmrErrorA;
-wor reg_qTmrErrorA;
-wor id_qTmrErrorA;
+wire word_addr_qTmrErrorC;
+wire we_qTmrErrorC;
+wire w_err_qTmrErrorC;
+wire valid_qTmrErrorC;
+wire req_qTmrErrorC;
+wire reg_qTmrErrorC;
+wire id_qTmrErrorC;
+wire word_addr_qTmrErrorB;
+wire we_qTmrErrorB;
+wire w_err_qTmrErrorB;
+wire valid_qTmrErrorB;
+wire req_qTmrErrorB;
+wire reg_qTmrErrorB;
+wire id_qTmrErrorB;
+wire word_addr_qTmrErrorA;
+wire we_qTmrErrorA;
+wire w_err_qTmrErrorA;
+wire valid_qTmrErrorA;
+wire req_qTmrErrorA;
+wire reg_qTmrErrorA;
+wire id_qTmrErrorA;
 logic [ObiCfg.DataWidth - 1:0] rsp_dataA;
 logic [ObiCfg.DataWidth - 1:0] rsp_dataB;
 logic [ObiCfg.DataWidth - 1:0] rsp_dataC;
