diff --git a/tmrg_files/obi_uartTMR.sv b/tmrg_files/obi_uartTMR.sv
index e077e93..7701607 100644
--- a/tmrg_files/obi_uartTMR.sv
+++ b/tmrg_files/obi_uartTMR.sv
@@ -40,7 +40,9 @@
  ****************************************************************************************************/
 
 module obi_uartTMR #(
-  parameter obi_pkg::obi_cfg_t ObiCfg = obi_pkg::ObiDefaultConfig
+  parameter obi_pkg::obi_cfg_t ObiCfg = obi_pkg::ObiDefaultConfig,
+  parameter type               obi_req_t = logic,
+  parameter type               obi_rsp_t = logic
 ) (
   input logic clk_iA,
   input logic clk_iB,
@@ -95,24 +97,24 @@ module obi_uartTMR #(
   output tmrErrorC
 );
 import obi_uart_pkg::*;
-wor i_uart_txtmrErrorC;
-wor i_uart_rxtmrErrorC;
-wor i_uart_registertmrErrorC;
-wor i_uart_modemtmrErrorC;
-wor i_uart_interruptstmrErrorC;
-wor i_uart_baudgentmrErrorC;
-wor i_uart_txtmrErrorB;
-wor i_uart_rxtmrErrorB;
-wor i_uart_registertmrErrorB;
-wor i_uart_modemtmrErrorB;
-wor i_uart_interruptstmrErrorB;
-wor i_uart_baudgentmrErrorB;
-wor i_uart_txtmrErrorA;
-wor i_uart_rxtmrErrorA;
-wor i_uart_registertmrErrorA;
-wor i_uart_modemtmrErrorA;
-wor i_uart_interruptstmrErrorA;
-wor i_uart_baudgentmrErrorA;
+wire i_uart_txtmrErrorC;
+wire i_uart_rxtmrErrorC;
+wire i_uart_registertmrErrorC;
+wire i_uart_modemtmrErrorC;
+wire i_uart_interruptstmrErrorC;
+wire i_uart_baudgentmrErrorC;
+wire i_uart_txtmrErrorB;
+wire i_uart_rxtmrErrorB;
+wire i_uart_registertmrErrorB;
+wire i_uart_modemtmrErrorB;
+wire i_uart_interruptstmrErrorB;
+wire i_uart_baudgentmrErrorB;
+wire i_uart_txtmrErrorA;
+wire i_uart_rxtmrErrorA;
+wire i_uart_registertmrErrorA;
+wire i_uart_modemtmrErrorA;
+wire i_uart_interruptstmrErrorA;
+wire i_uart_baudgentmrErrorA;
 logic rxdA;
 logic rxdB;
 logic rxdC;
