diff --git a/tmrg_files/obi_err_sbrTMR.sv b/tmrg_files/obi_err_sbrTMR.sv
index 1ddcd2b..9df18fb 100644
--- a/tmrg_files/obi_err_sbrTMR.sv
+++ b/tmrg_files/obi_err_sbrTMR.sv
@@ -41,6 +41,8 @@
 
 module obi_err_sbrTMR #(
   parameter obi_pkg::obi_cfg_t ObiCfg = obi_pkg::ObiDefaultConfig,
+  parameter type                         obi_req_t   = logic,
+  parameter type                         obi_rsp_t   = logic,
   parameter int unsigned NumMaxTrans = 1,
   parameter logic [ObiCfg.DataWidth - 1:0] RspData = 32'hBADCAB1E
 ) (
@@ -63,9 +65,9 @@ module obi_err_sbrTMR #(
   output tmrErrorB,
   output tmrErrorC
 );
-wor i_id_fifotmrErrorC;
-wor i_id_fifotmrErrorB;
-wor i_id_fifotmrErrorA;
+wire i_id_fifotmrErrorC;
+wire i_id_fifotmrErrorB;
+wire i_id_fifotmrErrorA;
 logic [ObiCfg.IdWidth - 1:0] ridA;
 logic [ObiCfg.IdWidth - 1:0] ridB;
 logic [ObiCfg.IdWidth - 1:0] ridC;
