diff --git a/tmrg_files/obi_xbarTMR.sv b/tmrg_files/obi_xbarTMR.sv
index 32fc38a..2af6fe0 100644
--- a/tmrg_files/obi_xbarTMR.sv
+++ b/tmrg_files/obi_xbarTMR.sv
@@ -42,10 +42,17 @@
 module obi_xbarTMR #(
   parameter obi_pkg::obi_cfg_t SbrPortObiCfg = obi_pkg::ObiDefaultConfig,
   parameter obi_pkg::obi_cfg_t MgrPortObiCfg = SbrPortObiCfg,
+  parameter type               sbr_port_obi_req_t = logic,
+  parameter type               sbr_port_a_chan_t  = logic,
+  parameter type               sbr_port_obi_rsp_t = logic,
+  parameter type               sbr_port_r_chan_t  = logic,
+  parameter type               mgr_port_obi_req_t = sbr_port_obi_req_t,
+  parameter type               mgr_port_obi_rsp_t = sbr_port_obi_rsp_t,
   parameter int unsigned NumSbrPorts = 32'd0,
   parameter int unsigned NumMgrPorts = 32'd0,
   parameter int unsigned NumMaxTrans = 32'd0,
   parameter int unsigned NumAddrRules = 32'd0,
+  parameter type               addr_map_rule_t    = logic,
   parameter bit UseIdForRouting = 1'b0,
   parameter bit [NumSbrPorts - 1:0] [NumMgrPorts - 1:0] Connectivity = '1
 ) (
@@ -83,15 +90,15 @@ module obi_xbarTMR #(
   output tmrErrorB,
   output tmrErrorC
 );
-wor i_muxtmrErrorC;
-wor i_err_sbrtmrErrorC;
-wor i_demuxtmrErrorC;
-wor i_muxtmrErrorB;
-wor i_err_sbrtmrErrorB;
-wor i_demuxtmrErrorB;
-wor i_muxtmrErrorA;
-wor i_err_sbrtmrErrorA;
-wor i_demuxtmrErrorA;
+wire [NumMgrPorts-1:0] i_muxtmrErrorC;
+wire [NumSbrPorts-1:0] i_err_sbrtmrErrorC;
+wire [NumSbrPorts-1:0] i_demuxtmrErrorC;
+wire [NumMgrPorts-1:0] i_muxtmrErrorB;
+wire [NumSbrPorts-1:0] i_err_sbrtmrErrorB;
+wire [NumSbrPorts-1:0] i_demuxtmrErrorB;
+wire [NumMgrPorts-1:0] i_muxtmrErrorA;
+wire [NumSbrPorts-1:0] i_err_sbrtmrErrorA;
+wire [NumSbrPorts-1:0] i_demuxtmrErrorA;
 logic [NumSbrPorts - 1:0] [cf_math_pkg::idx_width(NumMgrPorts) - 1:0] sbr_port_selectA;
 logic [NumSbrPorts - 1:0] [cf_math_pkg::idx_width(NumMgrPorts) - 1:0] sbr_port_selectB;
 logic [NumSbrPorts - 1:0] [cf_math_pkg::idx_width(NumMgrPorts) - 1:0] sbr_port_selectC;
@@ -110,7 +117,7 @@ sbr_port_obi_rsp_t [NumMgrPorts - 1:0] [NumSbrPorts - 1:0] mgr_rspsC;
 for(genvar i = 0; i<NumSbrPorts; i++)
   begin : gen_demux
 
-    addr_decodeTMR #(.NoIndices(NumMgrPorts), .NoRules(NumAddrRules), .rule_t(addr_map_rule_t)) i_addr_decode (
+    addr_decodeTMR #(.NoIndices(NumMgrPorts), .NoRules(NumAddrRules), .addr_t    ( logic [MgrPortObiCfg.AddrWidth-1:0] ), .rule_t(addr_map_rule_t)) i_addr_decode (
         .addr_iA(sbr_ports_req_iA[i].a.addr),
         .addr_iB(sbr_ports_req_iB[i].a.addr),
         .addr_iC(sbr_ports_req_iC[i].a.addr),
@@ -156,9 +163,9 @@ for(genvar i = 0; i<NumSbrPorts; i++)
         .mgr_ports_rsp_iA(sbr_rspsA[i]),
         .mgr_ports_rsp_iB(sbr_rspsB[i]),
         .mgr_ports_rsp_iC(sbr_rspsC[i]),
-        .tmrErrorA(i_demuxtmrErrorA),
-        .tmrErrorB(i_demuxtmrErrorB),
-        .tmrErrorC(i_demuxtmrErrorC)
+        .tmrErrorA(i_demuxtmrErrorA[i]),
+        .tmrErrorB(i_demuxtmrErrorB[i]),
+        .tmrErrorC(i_demuxtmrErrorC[i])
       );
   end
 for(genvar i = 0; i<NumSbrPorts; i++)
@@ -173,6 +180,9 @@ for(genvar i = 0; i<NumSbrPorts; i++)
             assign sbr_rspsA[i][j] = mgr_rspsA[j][i];
             assign sbr_rspsB[i][j] = mgr_rspsB[j][i];
             assign sbr_rspsC[i][j] = mgr_rspsC[j][i];
+            assign i_err_sbrtmrErrorA[i] = '0;
+            assign i_err_sbrtmrErrorB[i] = '0;
+            assign i_err_sbrtmrErrorC[i] = '0;
           end
 
         else
@@ -218,9 +228,9 @@ for(genvar i = 0; i<NumSbrPorts; i++)
                 .obi_rsp_oA(sbr_rspsA[i][j]),
                 .obi_rsp_oB(sbr_rspsB[i][j]),
                 .obi_rsp_oC(sbr_rspsC[i][j]),
-                .tmrErrorA(i_err_sbrtmrErrorA),
-                .tmrErrorB(i_err_sbrtmrErrorB),
-                .tmrErrorC(i_err_sbrtmrErrorC)
+                .tmrErrorA(i_err_sbrtmrErrorA[i]),
+                .tmrErrorB(i_err_sbrtmrErrorB[i]),
+                .tmrErrorC(i_err_sbrtmrErrorC[i])
               );
           end
       end
@@ -250,13 +260,13 @@ for(genvar i = 0; i<NumMgrPorts; i++)
         .mgr_port_rsp_iA(mgr_ports_rsp_iA[i]),
         .mgr_port_rsp_iB(mgr_ports_rsp_iB[i]),
         .mgr_port_rsp_iC(mgr_ports_rsp_iC[i]),
-        .tmrErrorA(i_muxtmrErrorA),
-        .tmrErrorB(i_muxtmrErrorB),
-        .tmrErrorC(i_muxtmrErrorC)
+        .tmrErrorA(i_muxtmrErrorA[i]),
+        .tmrErrorB(i_muxtmrErrorB[i]),
+        .tmrErrorC(i_muxtmrErrorC[i])
       );
   end
-assign tmrErrorA = i_demuxtmrErrorA|i_err_sbrtmrErrorA|i_muxtmrErrorA;
-assign tmrErrorB = i_demuxtmrErrorB|i_err_sbrtmrErrorB|i_muxtmrErrorB;
-assign tmrErrorC = i_demuxtmrErrorC|i_err_sbrtmrErrorC|i_muxtmrErrorC;
+assign tmrErrorA = (|i_demuxtmrErrorA)|( |i_err_sbrtmrErrorA )|( |i_muxtmrErrorA );
+assign tmrErrorB = (|i_demuxtmrErrorB)|( |i_err_sbrtmrErrorB )|( |i_muxtmrErrorB );
+assign tmrErrorC = (|i_demuxtmrErrorC)|( |i_err_sbrtmrErrorC )|( |i_muxtmrErrorC );
 endmodule
 
