diff --git a/tmrg_files/obi_uartTMR.sv b/tmrg_files/obi_uartTMR.sv
index 5a782cd..6fb7cfb 100644
--- a/tmrg_files/obi_uartTMR.sv
+++ b/tmrg_files/obi_uartTMR.sv
@@ -40,7 +40,9 @@
  ****************************************************************************************************/
 
 module obi_uartTMR #(
-  parameter obi_pkg::obi_cfg_t ObiCfg = obi_pkg::ObiDefaultConfig
+  parameter obi_pkg::obi_cfg_t ObiCfg = obi_pkg::ObiDefaultConfig,
+  parameter type               obi_req_t = logic,
+  parameter type               obi_rsp_t = logic
 ) (
   input logic clk_iA,
   input logic clk_iB,
