diff --git a/tmrg_files/periph_to_regTMR.sv b/tmrg_files/periph_to_regTMR.sv
index 5760d1d..9a7ad15 100644
--- a/tmrg_files/periph_to_regTMR.sv
+++ b/tmrg_files/periph_to_regTMR.sv
@@ -43,7 +43,9 @@ module periph_to_regTMR #(
   parameter int unsigned AW = 32,
   parameter int unsigned DW = 32,
   parameter int unsigned BW = 8,
-  parameter int unsigned IW = 0
+  parameter int unsigned IW = 0,
+  parameter type         req_t = logic,
+  parameter type         rsp_t = logic
 ) (
   input logic clk_iA,
   input logic clk_iB,
@@ -94,18 +96,18 @@ module periph_to_regTMR #(
   output tmrErrorB,
   output tmrErrorC
 );
-wor r_valid_qTmrErrorC;
-wor r_rdata_qTmrErrorC;
-wor r_opc_qTmrErrorC;
-wor r_id_qTmrErrorC;
-wor r_valid_qTmrErrorB;
-wor r_rdata_qTmrErrorB;
-wor r_opc_qTmrErrorB;
-wor r_id_qTmrErrorB;
-wor r_valid_qTmrErrorA;
-wor r_rdata_qTmrErrorA;
-wor r_opc_qTmrErrorA;
-wor r_id_qTmrErrorA;
+wire r_valid_qTmrErrorC;
+wire r_rdata_qTmrErrorC;
+wire r_opc_qTmrErrorC;
+wire r_id_qTmrErrorC;
+wire r_valid_qTmrErrorB;
+wire r_rdata_qTmrErrorB;
+wire r_opc_qTmrErrorB;
+wire r_id_qTmrErrorB;
+wire r_valid_qTmrErrorA;
+wire r_rdata_qTmrErrorA;
+wire r_opc_qTmrErrorA;
+wire r_id_qTmrErrorA;
 logic [IW - 1:0] r_id_dA;
 logic [IW - 1:0] r_id_dB;
 logic [IW - 1:0] r_id_dC;
