diff --git a/tmrg_files/soc_ctrl_reg_topTMR.sv b/tmrg_files/soc_ctrl_reg_topTMR.sv
index f5b80eb..b93a6a2 100644
--- a/tmrg_files/soc_ctrl_reg_topTMR.sv
+++ b/tmrg_files/soc_ctrl_reg_topTMR.sv
@@ -40,6 +40,8 @@
  ****************************************************************************************************/
 
 module soc_ctrl_reg_topTMR #(
+  parameter type reg_req_t = logic,
+  parameter type reg_rsp_t = logic,
   parameter int AW = 5,
   parameter int unsigned BootAddrDefault = 32'h0
 ) (
@@ -71,21 +73,21 @@ module soc_ctrl_reg_topTMR #(
 import soc_ctrl_reg_pkg::*;
 localparam int DW = 32;
 localparam int DBW = DW / 8;
-wor u_sram_dlytmrErrorC;
-wor u_fetchentmrErrorC;
-wor u_corestatustmrErrorC;
-wor u_bootmodetmrErrorC;
-wor u_bootaddrtmrErrorC;
-wor u_sram_dlytmrErrorB;
-wor u_fetchentmrErrorB;
-wor u_corestatustmrErrorB;
-wor u_bootmodetmrErrorB;
-wor u_bootaddrtmrErrorB;
-wor u_sram_dlytmrErrorA;
-wor u_fetchentmrErrorA;
-wor u_corestatustmrErrorA;
-wor u_bootmodetmrErrorA;
-wor u_bootaddrtmrErrorA;
+wire u_sram_dlytmrErrorC;
+wire u_fetchentmrErrorC;
+wire u_corestatustmrErrorC;
+wire u_bootmodetmrErrorC;
+wire u_bootaddrtmrErrorC;
+wire u_sram_dlytmrErrorB;
+wire u_fetchentmrErrorB;
+wire u_corestatustmrErrorB;
+wire u_bootmodetmrErrorB;
+wire u_bootaddrtmrErrorB;
+wire u_sram_dlytmrErrorA;
+wire u_fetchentmrErrorA;
+wire u_corestatustmrErrorA;
+wire u_bootmodetmrErrorA;
+wire u_bootaddrtmrErrorA;
 logic reg_weA;
 logic reg_weB;
 logic reg_weC;
